### PR TITLE
feat: configuration of PostgreSQL ident map file

### LINKED ISSUE #3362: [Feature]: allowing additional lines to be added to pg_ident file
### Is there an existing issue already for this feature request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new feature request to be evaluated.

### What problem is this feature going to solve? Why should it be added?

It would be nice to have a way to add user maps, so we can use certificate based authentication where the common name of the certificate is different from the postgres role name.

For example:

In pg_hba.conf:
`hostssl  all  +users  all  cert  map=user-map`

In pg_ident.conf
`user-map  /^(.*)\.user\.example\.com  \1`

There's a related issue where a certificate issuer I'm using doesn't allow the `_` in `streaming_replica`, which could be solved by mapping `streaming-replica` to `streaming_replica`.

### Describe the solution you'd like

My preferred solution would be to add a `pg_ident` list to the Custer CRD, where each line adds a line to the pg_ident file. This mirrors the current way to customize the pg_hba file.

### Describe alternatives you've considered

An alternative way could be to the manual addition of a `pg_ident.conf` file as volumeMount and allow to override the `ident_file` postgres setting, which is currently blocked from being modified.

### Additional context

No other context.

### Backport?

No

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by nilsbrenkman**
As described in the issue, we would like a way to add lines to the pg_ident.conf file.

I followed the same approach as pg_hba, it's basically a 1 on 1 mapping.

The pull request contains unit test and e2e test. Documentation needs to be added still.

Closes #3362

### CODE REVIEW

**Comment by armru** [comment]
Hello @nilsbrenkman We will look into this ASAP

File: pkg/management/postgres/configuration.go [hunk 1]
```diff
@@ -197,6 +198,42 @@ func quoteHbaLiteral(literal string) string {
...
+		pgIdentContent,
+		constants.PostgresqlIdentFile)
+	if err != nil {
+		return postgresIdentChanged, fmt.Errorf(
+			"installing postgresql Ident rules: %w",
+			err)
+	}
+
+	return postgresIdentChanged, err
+}
```
Comment (mnencia) [inline pkg/management/postgres/configuration.go:235]: I think that this code overlaps with the [WritePostgresUserMaps](https://github.com/cloudnative-pg/cloudnative-pg/blob/main/pkg/management/postgres/ident.go#L31) function. We should really have only one place doing this work.

**Review by armru — APPROVED** [review]

**Comment by armru** [comment]
Hello @nilsbrenkman , can you add the documentation?

**Review by mnencia — APPROVED** [review]

**Comment by nilsbrenkman** [comment]
> Hello @nilsbrenkman , can you add the documentation?

Will do!

**Pull request merged**